pkg/cloudprovider/providers/vsphere/vclib/virtualmachine.go

```diff
@@ -202,6 +202,27 @@ func (vm *VirtualMachine) IsActive(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
+// Exists checks if VM exists and is not terminated
+func (vm *VirtualMachine) Exists(ctx context.Context) (bool, error) {
+	vmMoList, err := vm.Datacenter.GetVMMoList(ctx, []*VirtualMachine{vm}, []string{"summary"})
```

Comment:
Since powerState is the only property used below, you reduce the response payload by specifying `summary.runtime.powerState`

Reply:
fixed